src/gluonts/evaluation/backtest.py

```diff
@@ -201,7 +202,7 @@ def backtest_metrics(
     )
 
     agg_metrics, item_metrics = evaluator(
-        ts_it, forecast_it, num_series=len(test_dataset)
+        ts_it, forecast_it, num_series=maybe_len(test_dataset)
```

Comment:
Why is this needed?

What happens if it returns `None`?

Reply:
Same as above: `len` may not be available. In `Evaluator.__call__` the `num_series` argument is `Optional`, and is only used for displaying progress: if a dataset has no `len`, the progress will be shown only partially (no percentage wrt total, only number of processed items).